swift.en.md

```diff
@@ -0,0 +1,340 @@
+- **[MUST]** Do not use `Void` as returning type.
+
+```swift
+// Bad
+func update() -> Void {
+}
+
+// Good
+func update() {
+}
+```
+
+## Closure
+
+- **[MUST]** Use trailing closure if the closure has just one argument.
+
+```swift
+// Bad
+client.search(keyword: "Sushi", completion: { recipes in
+})
+
+// Good
+client.search(keyword: "Sushi") { recipes in
+}
+```
+
+- **[MUST]** Do not use `()` when calling the closure has just one argument.
+
+```swift
+// Bad
+let titles = recipes.map() { $0.title }
+
+// Good
+let titles = recipes.map { $0.title }
+```
+
+- **[MUST]** Use shorthand argument if closure defined in one line.
+    - In contrast, Do not use it if closure defined in multiple lines.
+
+```swift
+// Bad
+let titles = recipes.map { recipe in recipe.title }
+
+// Good
+let titles = recipes.map { $0.title }
+```
+
+```swift
+// Bad
+client.search(keyword: "Sushi") {
+    if $1 != nil {
+        recipes = $0
+    }
+}
+
+// Good
+client.search(keyword: "Sushi") { results, error in
+    if error != nil {
+        recipes = results
+    }
+}
+```
+
+- **[MUST]** Do not use `@escaping` without needed
+- **[MUST]** Use `weak` instead of `unowned` as capture controls.
+    - In some cases, `unowned` improves performance. In other hand readability will be lost.
+
+## Property, Variable
+
+- **[MUST]** Use `let` to make variables immutable as possible.
+- **[MUST]** Drop type definition as possible. However, in the case of avoiding type inference for improvement compilation time, you can define clearly.
+
+### Property
+
+- **[MUST]** Use closest access control as possible.
+- **[MUST]** Use `private(set)` for the property want to write from limited scopes.
+- **[MUST]** Use `weak` as property attributes as possible.
+    - You should use `weak` to avoid circular references (eg. `@IBOutlet`, delegate)
+- **[MUST]** Do not use `unowned` as property attribute.
+    - In some cases, `unowned` improves performance. In other hand readability will be lost.
+
+```swift
+// Bad
+var delegate: FooDelegate
+@IBOutlet var textLabel: UILabel!
+
+// Good
+weak var delegate: FooDelegate
+@IBOutlet weak var textLabel: UILabel!
+```
+
+- **[MUST]** Do not use `get` block as possible on computed properties.
+
+```swift
+// Bad
+var foodCount: Int {
+    get { return foods.count }
+}
+
+// Good
+var foodCount: Int { 
+    return foods.count 
+}
+```
+
+- **[SHOULD]** If setting a property has a side-effect, write it in `willSet` or `didSet`.
+
+### Variable
+
+- **[MUST]** Do not use global variables.
+
+
+## Type
+
+- **[MUST]** Do not use `Array<T>`, `Dictionary<K, V>`. Use syntax sugar instead.
+
+```swift
+// Bad
+let seasonings: Array<String> = ["sugar", "salt", "vinegar", "soy sauce", "miso"]
+let ingredients: Dictionary<String, String> = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+
+// Good let seasonings: [String] = ["sugar", "salt", "vinegar", "soy sauce", "miso"] let ingredients: [String: String] = [ "bacon": "100g",
```

Comment:
Are we missing one (and very important line break)? 
Also the fact that we need to specify the type like `: [String]` is kinda breaking the rule above.